### PR TITLE
3080 ES flags and related Solr code clean up

### PR DESCRIPTION
**Pull request by albertisfu**
This PR addresses the second requirement mentioned in https://github.com/freelawproject/courtlistener/issues/3080 for removing flags and switches used during the transition from Solr to ES.

This PR depends on https://github.com/freelawproject/courtlistener/pull/4790 which should be merged first.

In this PR, all code related to the following flags and switches has been removed:

**Flags:**
```
r-es-search-api-active
o-es-search-api-active
o-es-active
oa-es-activate
r-es-active
p-es-active
oa-es-active
```

Switches:
```
o-es-alerts-active
es_resolve_citations
oa-es-alerts-active
```

Additionally, related Solr code that is no longer needed after the removal of these waffles has also been removed.

However, there are still some Solr methods that need to be removed. This could be addressed in a future PR once [#4767](https://github.com/freelawproject/courtlistener/pull/4767) is merged and also the removal of Solr indexing code that Eduardo is working on is merged.

Also, some commands that we decided not to migrate to ES still depend on related Solr code. We'll need to decide how to handle these if we want to remove the corresponding Solr code. For instance:

```
buried_alive_project
legal_robot
invoice_project
mfb
pharmaceutical_project
```

### CODE REVIEW

**Comment by mlissner** [comment]
Yaaaas!

File: cl/citations/match_citations.py [hunk 1]
```diff
@@ -200,14 +66,9 @@ def resolve_fullcase_citation(
...
-        if waffle.switch_is_active("es_resolve_citations"):
-            # Revolve citations using ES; enable once all the opinions are
-            # indexed.
-            db_search_results, _ = es_search_db_for_full_citation(
-                full_citation
-            )
-        else:
-            db_search_results = search_db_for_fullcitation(full_citation)
+        # Revolve citations using ES; enable once all the opinions are
+        # indexed.
```
Comment (mlissner) [inline cl/citations/match_citations.py:70]: We can remove this comment.

File: cl/search/feeds.py [hunk 2]
```diff
@@ -98,72 +93,38 @@ def items(self, obj):
...
+                    document_text_key = "plain_text"
+                    es_search_query = ESRECAPDocument.search()
+                    override_params = {
+                        "order_by": "entry_date_filed_feed desc",
+                    }
+                    exclude_docs_for_empty_field = "entry_date_filed"
+                case _:
+                    return []
+
+            # Do an ES query.
```
Comment (mlissner) [inline cl/search/feeds.py:118]: Can remove this comment too.

File: cl/search/templates/feeds/solr_desc_template.html
Comment (mlissner) [inline cl/search/templates/feeds/solr_desc_template.html:1]: Should we rename this file?

**Review by mlissner — APPROVED** [review]
I saw a few little things, but this was a really fun PR to review! Let's get one more quick set of eyes, and see what Eduardo can find that I missed.

File: cl/search/api_utils.py [hunk 3]
```diff
@@ -57,19 +54,13 @@ def get_object_list(request, cd, paginator, es_flag_status):
...
-        ]
-        and es_flag_status
-    )
+    is_oral_argument_active = cd["type"] == SEARCH_TYPES.ORAL_ARGUMENT
+    is_people_active = cd["type"] == SEARCH_TYPES.PEOPLE
+    is_opinion_active = cd["type"] == SEARCH_TYPES.OPINION
+    is_recap_active = cd["type"] in [
+        SEARCH_TYPES.RECAP,
+        SEARCH_TYPES.DOCKETS,
+    ]
```
Comment (ERosendo) [inline cl/search/api_utils.py:63]: A `match-case` structure could simplify this block and the following if statements.

File: cl/alerts/management/commands/cl_send_alerts.py [hunk 4]
```diff
@@ -232,13 +206,9 @@ def run_query(self, alert, rate, v1_webhook=False):
             qd["filed_after"] = cut_off_date
         elif query_type == SEARCH_TYPES.ORAL_ARGUMENT:
             qd["argued_after"] = cut_off_date
-            if waffle.switch_is_active("oa-es-alerts-active"):
-                # Return empty results for OA alerts. They are now handled
-                # by Elasticsearch.
-                return query_type, results, v1_results
```
Comment (ERosendo) [inline cl/alerts/management/commands/cl_send_alerts.py:238]: Shouldn't we keep the `return` statement? It appears that the previous behavior, when`oa-es-alerts-active` was active, was to simply return. This PR, however, removes that return statement.

**Review by ERosendo — APPROVED** [review]
Great work on the cleanup! Let's merge after addressing comments.

File: cl/alerts/management/commands/cl_send_alerts.py (hunk 4 above)
Comment (albertisfu) [reply cl/alerts/management/commands/cl_send_alerts.py:238]: Yeah! I got confused here, and this was a mistake. Thanks for catching it. I've refactored the command so it now only works for `SEARCH_TYPES.OPINION` alerts.

File: cl/search/feeds.py [hunk 5]
```diff
@@ -98,72 +93,38 @@ def items(self, obj):
         For Opinions SearchFeed returns clusters.
         For RECAP SearchFeed returns RECAPDocuments.
         """
-        search_form = SearchForm(obj.GET, request=obj)
+        search_form = SearchForm(obj.GET)
         if search_form.is_valid():
```
[the remark this answers is not in the document]
Comment (albertisfu) [reply cl/search/feeds.py]: Done!

File: cl/search/api_utils.py (hunk 3 above)
Comment (albertisfu) [reply cl/search/api_utils.py]: Yeah! I've applied the suggested refactor

File: cl/search/templates/feeds/solr_desc_template.html
Comment (albertisfu) [reply cl/search/templates/feeds/solr_desc_template.html:1]: sure, renamed.

File: cl/search/feeds.py (hunk 2 above)
Comment (albertisfu) [reply cl/search/feeds.py]: done!

File: cl/citations/match_citations.py (hunk 1 above)
Comment (albertisfu) [reply cl/citations/match_citations.py]: done!

**Comment by albertisfu** [comment]
I've applied the suggestions. Also removed the following commands:

```
buried_alive_project
legal_robot
invoice_project
mfb
pharmaceutical_project
```

Along with all their related Solr code.

There are now just a few Solr methods in `search_utils` left that can be removed once https://github.com/freelawproject/courtlistener/pull/4767 is merged.

This is ready for another review.

Thanks!

File: cl/search/forms.py [hunk 6]
```diff
@@ -467,15 +464,9 @@ def __init__(self, *args, **kwargs):
         fields dict.
         """
 
-        # Default values for Solr version.
-        default_status = "Precedential"
-        status_index = 1
-        if request and waffle.flag_is_active(request, "o-es-active"):
```
Comment (albertisfu) [inline cl/search/forms.py:473]: @ERosendo here is an example

**Pull request merged**